File: 0601-0700/0696-Count Binary Substrings/0696-Count Binary Substrings.cpp
class Solution {
public:
    int countBinarySubstrings(string s) {
        int n = s.size();
        int i = 0;
        int prev = 0;
        int count = 0;
        while (i < n) {
            int curr = 0;
            char target = s[i];
            while (i < n && s[i] == target) {
                ++curr;
                ++i;
            }
            count += min(prev, curr);
            prev = curr;
        }
        return count;
    }
};

class Solution {
public:
    int countBinarySubstrings(string s) {
        int prev = 0, curr = 1;
        int count = 0;
        for (int i = 1; i < s.size(); ++i) {
            if (s[i] != s[i - 1]) {
                count += min(prev, curr);
                prev = curr;
                curr = 1;
            }
            else {
                ++curr;
            }
        }
        return count + min(prev, curr);
    }
};

File: 0601-0700/0696-Count Binary Substrings/0696-Count Binary Substrings.py
class Solution:
    def countBinarySubstrings(self, s: str) -> int:
        count, prev, curr = 0, 0, 1
        for i in range(1, len(s)):
            if s[i] != s[i - 1]:
                count += min(prev, curr)
                prev = curr
                curr = 1
            else:
                curr += 1
        return count + min(prev, curr)
